kmod/src/inode.c

```diff
@@ -1595,10 +1602,12 @@ int scoutfs_scan_orphans(struct super_block *sb)
 			ret = -ENOENT;
 			break;
 		}
+
+		deleted++;
```

Comment:
I know it sounds crazy, but I don't see anything that stops it from being possible to delete billions of files.  We're in that world.  Just make it a return 1 if it made progress, the caller doesn't care about the count at all.  And while we're at it maybe limit this to deleting a few thousand files per scheduled work?  It'll keep chain scheduling while there's items, but we want unmount to be able to get in without it having to chew through a potentially enormous amount of orphan items.

Reply:
Renaming scan_orphans to delete_orphans.

OK. I'm going to add a param to the fn for max orphans to delete, because I think we want that value coming from the oino code, not in this fn or inode.c in general. Inital max value 4096.